relm-gen-widget/src/lib.rs

```diff
@@ -341,13 +343,62 @@ impl Driver {
     }
 
     fn get_view(&mut self, name: &Ident, typ: &Ty) -> View {
-        {
-            let segments = &self.view_macro.as_ref().expect("view! macro missing").path.segments;
-            if segments.len() != 1 || segments[0].ident != "view" {
-                panic!("Unexpected macro item")
+        // This method should probably just be replaced with `impl_view` and
+        // `view_validation_before_impl` should be put inside `impl_view`
+        self.view_validation_before_impl();
+        self.impl_view(name, typ)
+    }
+
+    fn view_validation_before_impl(&mut self) {
+        // This is what comes immediately after `view!` e.g. `{ ... }`
+        let macro_token_tree = &self.view_macro.as_ref().expect("`view!` macro not yet set").tts;
+        // Panic if the macro is declared as anything other than `view! { ... }` or equivalent
+        if macro_token_tree.len() != 1 {
+            panic!("Invalid `view!` syntax, must be `view! { ... }`, `view! ( ... )`, or `view! [ ... ]`");
+        }
+        // Reach inside the brackets and bind the contents (the top level items) of `view!`
+        let top_level_items = match &macro_token_tree[0] {
+            &TokenTree::Delimited(Delimited {ref tts, ..}) => tts.clone(),
+            _ => panic!("Contents of `view!` should be a comma-delimitted series of items")
+        };
+        if top_level_items.len() > 1 {
+            // If the token tree isn't empty, count the number of items at the top level of `view!`
+            let mut comma_previously_found = false;
+            for item in top_level_items {
```

Comment:
I think we can replace this big loop by:
```rust
if let Some(index) = top_level_items.iter().position(|item| item == TokenTree::Token(Token::Comma)) {
    if index != top_level_items.len() - 1 {
        panic!("…");
    }
}

Reply:
Done